apps/examples/nettest/nettest_stress.c

```diff
@@ -0,0 +1,376 @@
+				continue;
+			}
+			nbytes_sum = 0;
+
+			count++;
+			printf("[RECV] - Received Msg # %d] Received Msg (%d)\n", count, nbytes_sum);
+			if (count == PKT_NUM) {
+				sem_post(&g_sem_signal);
+				count = 0;
+			}
+		}
+	}
+	printf("[RECV] Closed successfully \n");
+	sem_post(&g_sem_signal);
+	close(g_sockfd);
+
+	return 0;
+}
+
+void *disconnector_thread(void *arg)
+{
+	printf("[DISC] SHUTDOWN START\n");
+	shutdown(g_sockfd, SHUT_RDWR);
+	printf("[DISC] SHUTDOWN DONE\n");
+#if PRINT_HEAPINFO
+	disconnect_count++;
+	if (disconnect_count == 10) {
+		heapinfo_parse(BASE_HEAP, HEAPINFO_SIMPLE, HEAPINFO_PID_ALL);
+		disconnect_count = 0;
+	}
+#endif
+	terminate_flag = 1;
+	return 0;
+}
+
+void nettest_stress(char *t_addr, int port)
+{
+	pthread_t sender;
+	pthread_t receiver;
+	pthread_t disconnector;
+	int ret = 0;
+	int iter = 0;
+	int connect_trial = 0;
+	int addrlen = sizeof(struct sockaddr_in);
+
+#if WIFI_AUTO_CONNECT
+	/* Connect Info
+	 * SSID : Gorani
+	 * PASS : jonbeo1@
+	 */
+	wifi_manager_result_e wret = WIFI_MANAGER_FAIL;
+	wret = wifi_manager_init(&wifi_callbacks);
+	if (wret != WIFI_MANAGER_SUCCESS) {
+		printf("WiFi connection failed\n");
+		return;
+	}
+
+	wifi_manager_ap_config_s config;
+	config.ssid_length = strlen(WIFI_SSID);
+	config.passphrase_length = strlen(WIFI_PASSPHRASE);
+	strncpy(config.ssid, WIFI_SSID, config.ssid_length + 1);
+	strncpy(config.passphrase, WIFI_PASSPHRASE, config.passphrase_length + 1);
+	config.ap_auth_type = 3;
+	config.ap_crypto_type = 3;
+	/* current wifi mode is station, then this try will succeed */
+	wret = wifi_manager_connect_ap(&config);
+	if (wret != WIFI_MANAGER_SUCCESS) {
+		printf("WiFi connection failed\n");
+		return;
+	}
+	WIFITEST_WAIT;
+	printf("Wi-Fi connection established\n");
+#endif
+
+	g_myaddr.sin_family = AF_INET;
+	g_myaddr.sin_port = htons(port);
+	inet_pton(AF_INET, t_addr, &(g_myaddr.sin_addr));
+
+	while (1) {
+		terminate_flag = 0;
+		retry_sender = 0;
+		retry_receiver = 0;
+		printf("\n\n[MAIN] Iteration START %d\n", iter + 1);
+		g_sockfd = socket(PF_INET, SOCK_STREAM, 0);
+		if (g_sockfd < 0) {
+			printf("[MAIN] TCP socket failure %d\n", errno);
+			return;
+		}
+
+		struct timeval tv;
+		tv.tv_sec = 1;
+		tv.tv_usec = 0;
+
+		if (setsockopt(g_sockfd, SOL_SOCKET, SO_RCVTIMEO, (struct timeval *)&tv, sizeof(struct timeval)) == -1) {
+			printf("[MAIN] rcv timeout set failed\n");
+			close(g_sockfd);
+			return;
+		}
+
+		printf("[MAIN] Connecting...\n");
+		if (connect(g_sockfd, (struct sockaddr *)&g_myaddr, addrlen) < 0) {
+			printf("[MAIN] connect %d-th fail: %d\n", ++connect_trial, errno);
+			close(g_sockfd);
+			sleep(1);
+			continue;
+		}
+		connect_trial = 0;
+		sem_init(&g_sem_signal, 0, 0);
+
+		int tmp = 7 + (rand() % 11);
+		printf("[MAIN] SHUTDOWN after %d\n", tmp);
+
+		ret = pthread_create(&sender, NULL, sender_thread, NULL);
+		if (ret != 0) {
+			printf("[MAIN] Failed to start sender_thread (%d)\n", ret);
+			return;
```

Comment:
In this case, you'd better destroy the `g_sem_signal`.

Reply:
Modified. Thanks for the comment.